src/providers/user/user-provider.js

```diff
@@ -92,6 +95,29 @@ export default class UserProvider extends EventEmitter {
         });
     }
 
+    async canSetMissionStatus() {
+        const user = await this.getCurrentUser();
+        const writeParameters = user.getWriteParameters();
+
+        const areParametersStatus = await Promise.all(
+            writeParameters.map(parameterName => isMissionStatusParameter(parameterName))
+        );
+
+        return areParametersStatus.some(isParameterStatus => isParameterStatus);
+    }
+
+    async getPossibleMissionStatuses() {
+        const possibleStatuses = await this.missionStatus.getAllMissionStatusRoles();
+
```

Comment:
Typo here: `await this.missionStatus.getAllMissionStatusRoles()`? Should be `await this.missionStatus.getPossibleMissionStatuses()`?

Reply:
@shefalijoshi this is still very much a WIP-- will be ready to review later today